crates/e2e/src/setup/services.rs

```diff
@@ -144,6 +144,8 @@ impl<'a> Services<'a> {
 
         let args = [
             "autopilot".to_string(),
+            "--run-loop-mode=sync-to-blockchain".to_string(),
+            "--max-run-loop-delay=400ms".to_string(),
```

Comment:
`400ms` seems a bit long. Did smaller values cause issues here?

Reply:
I think this is what happens:
1. begin initialization of autopilot
2. autopilot builds `CurrentBlockStream` which sets current block as latest block
3. autopilot finishes initialization
4. if `3` takes longer than `--max-run-loop-delay` autopilot considers the latest block too old and waits for the new one

Because the test code is blocked until the autopilot has reported it's healthy (which requires new blocks to be mined) we are stuck in a deadlock situation.
I think there are a few way to handle this:
1. continuously mine blocks in a background task while the autopilot is starting up until you see an auction being built
2. adjust run loop to skip waiting for a block if you pass for example `--max-run-loop-delay 0s`
3. detect that orderbook and autopilot are up without looking up the latest auction in the DB

Given that all tests so far assumed that the autopilot builds an auction with whatever onchain state was available so far `1` seems the most reasonable choice to me as it would change the least invariants.
That way `--max-run-loop-delay` could become very small and we don't a continuous block mining in the background which gives us better control over when we build auctions.

Reply:
By default anvil only mines blocks on demand (i.e. when submitting a transaction). It can be configured to do auto mining but I think having the tighter control of "manually" deciding when to mine a block can actually be nice for our tests.